src/library_manager/lib_manager.c

```diff
@@ -155,23 +191,20 @@ static void lib_manager_unload_aux(struct sof_man_module *mod,
 static int lib_manager_load_module(uint32_t module_id, struct sof_man_module *mod,
 				   struct sof_man_fw_desc *desc)
 {
-	struct ext_library *ext_lib = ext_lib_get();
 	uint32_t lib_id = LIB_MANAGER_GET_LIB_ID(module_id);
-	size_t load_offset = (size_t)((void *)ext_lib->desc[lib_id]);
+	struct lib_manager_mod_ctx *ctx = lib_manager_get_mod_ctx(module_id);
+	uint8_t *load_base = (uint8_t *)ctx->desc;
 	void __sparse_cache *va_base_text = (void __sparse_cache *)
 		mod->segment[SOF_MAN_SEGMENT_TEXT].v_base_addr;
-	void *src_txt = (void *)(mod->segment[SOF_MAN_SEGMENT_TEXT].file_offset + load_offset);
-	size_t st_text_size = mod->segment[SOF_MAN_SEGMENT_TEXT].flags.r.length;
+	void *src_txt = (void *)(load_base + mod->segment[SOF_MAN_SEGMENT_TEXT].file_offset);
+	size_t st_text_size = ctx->segment_size[SOF_MAN_SEGMENT_TEXT];
 	void __sparse_cache *va_base_rodata = (void __sparse_cache *)
 		mod->segment[SOF_MAN_SEGMENT_RODATA].v_base_addr;
-	void *src_rodata =
-		(void *)(mod->segment[SOF_MAN_SEGMENT_RODATA].file_offset + load_offset);
-	size_t st_rodata_size = mod->segment[SOF_MAN_SEGMENT_RODATA].flags.r.length;
+	void *src_rodata = (void *)(load_base +
+				    mod->segment[SOF_MAN_SEGMENT_RODATA].file_offset);
+	size_t st_rodata_size = ctx->segment_size[SOF_MAN_SEGMENT_RODATA];
 	int ret;
 
```

Comment:
Page alignment for sections is caused by secirity reasons. TEXT should be read_only while BSS read/write. Additionally if we want to introduce module level access rights, the PAGE size is minimal quantity we can control.